### ISSUE TITLE
Installation instructions need to start from the beginning

### ISSUE THREAD

**Issue by leonerd** [nvim-treesitter/nvim-treesitter#5519]
I've been struggling to work out how to install nvim-treesitter on a fresh neovim install. Your "installation" page is simply a list of what steps to use for a variety of plugin managers. As a new user I don't even know what is a plugin manager - do I have one already? which one do I pick? If not how do I get one?

With much help and handholding from `#neovim:matrix.org` I managed to work it out. I'd have got on much easier if the docs started with some words like this:

---

# Installation

`nvim-treesitter` is not shipped as part of neovim itself, so it needs to be installed as a plugin. Installing a plugin itself requires a plugin manager. Below is a list of instructions on how it can be installed depending on your chosen plugin manager. If you do not yet have a plugin manager we currently recommend `lazy` as it seems to be the simplest to get working.

## Installing lazy

Full instructions can be found [here](https://github.com/folke/lazy.nvim#-installation); but in simple terms, you'll need to add some Lua config to nvim's startup, let it run once, and then run it again with the code to install nvim-treesitter.

If you are starting from a blank slate, create yourself a `init.lua` and paste in:

```lua
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    "--branch=stable", -- latest stable release
    lazypath,
  })
end
vim.opt.rtp:prepend(lazypath)
```

Note that you cannot have both a `init.vim` and `init.lua` file so if you already have some localised config in the former, you might wish to use an inline `lua` block in `init.vim` instead:

```vim
lua << EOF
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
...
end
vim.opt.rtp:prepend(lazypath)
EOF
```

At this point, quit and restart nvim. This will install `lazy` itself. Now that is installed, add an extra line to the end of the Lua code:

```lua
require("lazy").setup(
  {{"nvim-treesitter/nvim-treesitter", build = ":TSUpdate"}}
)
```

Now quit and restart nvim a second time and tree-sitter will be installed. You can now confirm this by checking the output of

```vim
:TSConfigInfo
```

The command should exist, and print some config settings. If this happens, you have successfully installed `nvim-treesitter.`

---

I would have sent you a PR to add these docs to the wiki directly, but for all my poking at the wiki page I coudn't see an edit button or a way to raise a PR against it. So perhaps this will have to do.

**Comment by clason** [comment]
Thank you! The rationale is that users are supposed to have basic knowledge of Neovim before they start installing _any_ plugins, much less a still experimental one such as nvim-treesitter. (You'll find most plugin READMEs assume this basic knowledge.)

It is indeed a good question where users are expected to get this information from; ideally this should be Neovim itself, but we do not want to bless any specific plugin manager (which are all external projects). We certainly do not want `lazy.nvim` to become the official plugin manager (nice as it is, it's much too heavy and opinionated for that).

**Comment by clason** [comment]
> Accept that you'll have to pick a compromise decision and just pick something they can start from.

Accept that this (not blessing plugin managers) is a reasoned (Neovim) project decision -- anything we officially endorse, we become responsible for. And people seem to manage well enough (though some indeed require more help than others).

Again, giving more pointers in _this_ readme is a well-made suggestion, and I will act on it.

**Comment by iMaCHVFX** [comment]
as a beginner who needs syntax highlighting and dont know any other alternatives in neovim, the installation instruction are some of the most confusing installation instructions i have read in my life.

stacking that on top of lazy's horrible documentation/instructions makes installing this plugin near impossible.
never in my life have i felt like i was taking 1 step forward and 10 steps back EVERY SINGLE STEP OF THE WAY.

something that you guys need to realize (this goes for everysingle developer it seems like cause documentation everywhere is horrible) is that not everyone that wants to use your product are going to have 10 years experience in development or neovim or whatever the category of thing is. its just an unrealistic expectation, even if unintentional that is how it comes across when you make things in this way.

at THE VERY LEAST create a section for people who are just starting out, i cannot tell you how impossible it is to get help with this sort of stuff so saying "some require more help than others" is just ignorant if you've spent even 10 minutes as a beginning trying to get help in the last 10 years, either 1 no replies, 2 they dont answer it at all and tell you to look at the documentation (and i have which is why im there asking for help in the first place), 3 they defend the documentation and call you dumb. i only get actual help every 1 in 10 times i ask a question literally anywhere. so maybe dont be so dismissive of other needing "a little more help" when that little more help doesnt exist anywhere amongst the development community, atleast not in an easily accessible place.

**Comment by clason** [comment]
> as a beginner who needs syntax highlighting and dont know any other alternatives in neovim,

As a beginner, you should not be using this plugin, period. Neovim has perfectly fine syntax highlighting built-in, and there's _nothing_ you have to do to enable that.

I cannot stress enough that **this plugin is experimental and not yet stable**.

**Comment by iMaCHVFX** [comment]
the is basically the exact opposite of what ive been told from everyone and every place that tries to help with setting up neovim they're ALL like "use treesitter use treesitter" (which is fantastic when you're not installing manually (used it with NvChad))

**Comment by clason** [comment]
Yes, and they should stop doing that. Alas, monetization wins over reason, and we can't do more than but a big fat warning in our README. (And, seriously, if you ignore that, why should we believe you'd be reading any other instructions we put in there? Not exactly motivating, you know.)

**Comment by skiss9** [comment]
I added an issue along the same lines but with simpler and updated install instructions for lazy.nvim: https://github.com/nvim-treesitter/nvim-treesitter/issues/7276

No need to write code to git clone, lazy.nvim does that automatically.

I'm using lazy.nvim at the moment but I agree that it's too heavy, and it still requires mason to do some of the work. Open to switching to something simpler.

I actually like how helix is handling a lot of that, if they manage to add a plugin system and a scripting language like lua I could see them replacing nvim.
